packages/backend/src/middlewares/validator.ts

```diff
@@ -0,0 +1,18 @@
+// import { HTTP400Error } from '../utils/httpErrors'
+import { Schema, ZodError } from 'zod'
+import { Request, Response, NextFunction } from 'express'
+import { BadRequestException } from '../errors/badRequestException'
+
+export function validate(schema: Schema) {
+  return (req: Request, _res: Response, next: NextFunction) => {
+    try {
+      schema.parse(req.body)
+      next()
+    } catch (error) {
+      if (error instanceof ZodError) {
+        throw new BadRequestException(error.errors[0].message)
+      }
+      next(error)
+    }
```

Comment:
We can avoid using `parse` since we will have to wrap everything in a `try-catch` block. I can suggest using `safeParse`:

```suggestion
    const result = schema.safeParse(req.body)
    
    if(!result.success) {
      throw new BadRequestException(error.errors[0].message)
    }
    
    next()
```

Reply:
🤔 I think by having a middleware that is validating a schema we would not get the types for the request's body in a controller.

Reply:
removed the validator and went for a different approach which is able to infer types. 
The validation will be done at the controller level, no longer a middleware that precedes the controller.